core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -984,20 +984,39 @@ private[spark] class SparkSubmit extends Logging {
         e
     }
 
+    //    this variable is used to judge whether driver pod is normal if spark is on k8s
+    var DriverPodIsNormal: Boolean = if (args.master.startsWith("k8s")) true else false
+    var driverThrow: Throwable = null
     try {
       app.start(childArgs.toArray, sparkConf)
     } catch {
       case t: Throwable =>
-        throw findCause(t)
+        logWarning("Some ERR/Exception happened when app is running.")
+        if (args.master.startsWith("k8s")) {
+          DriverPodIsNormal = false
+          driverThrow = t
+        } else {
+          throw findCause(t)
+        }
     } finally {
       if (args.master.startsWith("k8s") && !isShell(args.primaryResource) &&
           !isSqlShell(args.mainClass) && !isThriftServer(args.mainClass) &&
           !isConnectServer(args.mainClass)) {
         try {
+          logWarning("Begin to close SparkContext inside driver pod......")
           SparkContext.getActive.foreach(_.stop())
         } catch {
           case e: Throwable => logError(s"Failed to close SparkContext: $e")
-        }
+        } finally {
+          if (SparkContext.getActive.isEmpty) {
+            logWarning("Finished to close SparkContext inside driver pod successfully.")
+            if (!DriverPodIsNormal) {
+              logError(s"Driver Pod will exit because: $driverThrow")
+              System.exit(1)
```

Comment:
Please use the official `SparkExitCode`.

https://github.com/apache/spark/blob/a3c04ec1145662e4227d57cd953bffce96b8aad7/core/src/main/scala/org/apache/spark/util/SparkExitCode.scala#L23